gnovm/pkg/gnolang/gonative.go

```diff
@@ -824,19 +824,25 @@ func gno2GoType(t Type) reflect.Type {
 			panic("should not happen")
 		}
 	case *PointerType:
-		et := gno2GoType(ct.Elem())
+		et := gno2GoType(ct.Elem(), seen)
 		return reflect.PointerTo(et)
 	case *ArrayType:
 		ne := ct.Len
-		et := gno2GoType(ct.Elem())
+		et := gno2GoType(ct.Elem(), seen)
 		return reflect.ArrayOf(ne, et)
 	case *SliceType:
-		et := gno2GoType(ct.Elem())
+		et := gno2GoType(ct.Elem(), seen)
 		return reflect.SliceOf(et)
 	case *StructType:
+		if val, ok := seen[t]; ok {
+			return val
+		}
```

Comment:
don't know if this case will work ?

```
package main

import "fmt"

func main() {
	t := struct {
		Name string
		Next interface{}
	}{}
	t.Next = &t
	fmt.Printf("%v", t)
}
```

Reply:
No, but this case is a bit more tricky to handle and does not involve the same code block so i think it would be pertinent to work on it on a different PR